Sources/Endpoints/Request.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Request.swift
+//  Endpoints
+//
+//  Created by Simon Jarbrant on 2020-06-28.
+//
+
+import Foundation
+
+public struct Request {
+    private let _cancel: () -> Void
+
+    public init(cancel: @escaping () -> Void) {
+        self._cancel = cancel
+    }
+
+    func cancel() {
+        _cancel()
+    }
+}
+
+// MARK: - Convenience init with a URLSessionDataTask
+public extension Request {
+    init(dataTask: URLSessionDataTask) {
+        self.init {
+            dataTask.cancel()
+        }
+    }
```

Comment:
Just curious. Is there any concern for retain cycles? Would for example the `Request` retain the `URLSessionDataTask` even after it is completed? The implicit capture list should be `[datatask] in` with a strong reference if I remember correctly.

Reply:
@Ahlisen You are correct in that the `Request` will keep an implicit strong reference (via the closure) to the underlying `URLSessionDataTask`, so long as someone is keeping the request object itself around. There would be no retain cycle AFAICT though (since the `URLSessionDataTask` has no reference to the `Request`), unless I'm missing/forgetting something 🤔  

I was thinking maybe this could be used to (in the future) implement pause/resume behavior for requests, or maybe retrieving some details from the in-flight operation. Maybe it's then a better idea to _not_ capture the datatask in the closure but instead "hide" the datatask under a protocol, so that setting up `Request` objects doesn't require a datatask. I'm open to ideas there 🙂

Reply:
I agree on the name-change part of this pitch since Cancellable collides with the one in Combine (which is introduced in another PR). I'm just curious if you could give an example of a benefit of having it as a concrete type?